controllers/substitutions.js

```diff
@@ -22,12 +22,20 @@ const exec = require('child_process').exec;
 var Compendium = require('../lib/model/compendium');
 
 exports.view = (req, res) => {
-  console.log("req.query");
-  console.log(req.query);
   var answer = {};
   var filter = { "metadata.substituted": true };
   var limit = parseInt(req.query.limit || config.list_limit, 10);
   var start = parseInt(req.query.start || 1, 10) - 1;
+  
+  if (req.query.base) {
```

Comment:
Your are re-creating the filter object although that is not necessary. Imagine you want to change the default for "metadata.substituted" to `false`, that would need a change in **4** places, which is likely to go wrong.

See https://github.com/o2r-project/o2r-muncher/blob/master/controllers/compendium.js#L118 and following.

Reply:
Thanks for the helping Link. It's done now.